tests/test_dataclasses.py

```diff
@@ -276,3 +276,37 @@ class Reordered(Base):
         assert [p.name for p in params_base] == ["self", "a", "b"]
         assert [p.name for p in params_reordered] == ["self", "b", "c", "a"]
         assert str(params_reordered["b"].annotation) == "float"
+
+
+def test_parameters_annotated_as_initvar() -> None:
+    """Don't return InitVar annotated fields as class members.
+
+    But if __init__ is defined, InitVar has no effect.
+    """
+
+    code = """
+    from dataclasses import dataclass, InitVar
+
+    @dataclass
+    class PointA:
+        x: float
+        y: float
+        z: InitVar[float]
+
+    @dataclass
+    class PointB:
+        x: float
+        y: float
+        z: InitVar[float]
+
+        def __init__(self, r: float): ...
+    """
+
+    with temporary_visited_package("package", {"__init__.py": code}) as module:
+        PointA = module["PointA"]
+        assert ["self", "x", "y", "z"] == [p.name for p in PointA.parameters]
+        assert ["x", "y", "__init__"] == [name for name in PointA.members]
+
+        PointB = module["PointB"]
+        assert ["self", "r"] == [p.name for p in PointB.parameters]
+        assert ["x", "y", "z", "__init__"] == [name for name in PointB.members]
```

Comment:
Ideally `z` would be removed here too, but since `__init__` was already present we didn't apply the dataclasses logic so `z` wasn't removed. That's fine, lets just leave it like that: we are all consenting adults and users can shoot their own foot if they want to :slightly_smiling_face: